### DISCUSSION TITLE
Build specific subdirectory of project, instead of entire project

### DISCUSSION THREAD

**Discussion by raiguard** [crowdin/crowdin-cli#578] [Q&A] [answered]
Greetings,

My [Crowdin project](https://crowdin.com/project/raiguards-factorio-mods) has translations for multiple small projects combined into one project on Crowdin. Each project has a separate subfolder. My Crowdin CLI script runs during my release workflow, with a simple `crowdin push sources; crowdin pull` among some workarounds for INI quoting rules. You can view my script [here](https://github.com/raiguard/dotfiles/blob/master/bin/factorio-crowdin-sync).

Anyway, here's the issue: every time I make a mod release, Crowdin builds the entire project and sends the translations for every single mod. The CLI detects this and simply throws away all of the translations it doesn't need, which means it works fine, but I'm wasting a lot of Crowdin server time and building the entire project makes the release take way longer.

Is there a way to tell it to only build a specific subdirectory instead of the entire project?

**Comment by andrii-bodnar** [comment, marked as answer]
You can use [branches](https://support.crowdin.com/versions-management/) to organize your projects.

Then you can just call the `crowdin push sources` and `crowdin pull` by adding the `-b` parameter and CLI will push, build and download only the specified branch. For example:

```console
crowdin push -b project1
crowdin pull -b project1

...

crowdin push -b project2
crowdin pull -b project2
```

Additionally, this will allow auto-translating the duplicated strings between the projects in case there are some common strings.

**Reply by raiguard** [reply]
Thank you, I will give that a shot!
